system/jlib/jsocket.cpp

```diff
@@ -1244,7 +1244,7 @@ ISocket* CSocket::accept(bool allowcancel, SocketEndpoint *peerEp)
     }
 
     DEFINE_SOCKADDR(peerSockAddr);      // used if peerIp
-    socklen_t peerSockAddrLen = sizeof(peerSockAddr);
+    socklen_t peerSockAddrLen = sizeof(peerSockAddr.sa);
 
     T_SOCKET newsock;
```

Comment:
@mckellyln I don't understand this code.   The logic seems to be:
i) declare a union struct
ii) pass in the first member of the union
then call 
        getSockAddrEndpoint(peerSockAddr, peerSockAddrLen, *peerEp);
which either accesses the second or third elements of the union.  That includes extracting the port - which as far as I can tell has not been initialised.....
There are also 3 other places in the same file that call getSockAddrEndpoint() with similar patterns/problems.

Reply:
peerSockAddr is set in in the ::accept() call and should include the port.
In the other places u and ul should be set by the getpeername() or recvfrom() etc. calls.

Reply:
I might be inclined to use ul = sizeof(u.sin) in these places as that is probably more typical.
The internal struct is like:
```
using sockaddr_t = union {
    sockaddr_storage ss;
    sockaddr_in6     sin6;
    sockaddr_in      sin;
    sockaddr         sa;
};
```